### DISCUSSION TITLE
Support nested form params in http-client

### DISCUSSION THREAD

**Discussion by CambodianCoder** [babashka/http-client#27] [Ideas]
Noticed that this library doesn't appear to natively supported nested form params on requests.

For example, with this library, one could attempt to send a post request with form-params of `{"foo" {"bar" true}, "baz" [{"quix" 0 }]}`, but the server fails to decode the nested json correctly. However, sending form-params of the format `{"foo[bar]" true, "baz[0]" {"quix" 0 }` succeeds.

It looks like other clojure http libraries have implemented support for this, at
https://github.com/clj-commons/clj-http-lite/issues/43, and https://github.com/clj-commons/clj-http-lite/pull/44/files#diff-eca06351807a8cbec22e826d3ee44ae631af3a29af8b7cfc6734d5824f89afa2R233.

 I was wondering if babashka/http-client could also support nested form params natively?

**Comment by borkdude** [comment]
Sure, a Github issue + PR is welcome
